tools/tsp-client/src/git.ts

```diff
@@ -1,29 +1,18 @@
-import { execSync, spawn } from "child_process";
+import { spawn } from "child_process";
+import { simpleGit } from "simple-git";
 
-export function getRepoRoot(): string {
-    return execSync('git rev-parse --show-toplevel').toString().trim();
+export async function getRepoRoot(repoPath: string): Promise<string> {
+  return await simpleGit(repoPath).revparse(["--show-toplevel"]);
 }
-  
+
 export async function cloneRepo(rootUrl: string, cloneDir: string, repo: string): Promise<void> {
-    return new Promise((resolve, reject) => {
-      const git = spawn("git", ["clone", "--no-checkout", "--filter=tree:0", repo, cloneDir], {
-        cwd: rootUrl,
-        stdio: "inherit",
-      });
-      git.once("exit", (code) => {
-        if (code === 0) {
-          resolve();
-        } else {
-          reject(new Error(`git clone failed exited with code ${code}`));
-        }
-      });
-      git.once("error", (err) => {
-        reject(new Error(`git clone failed with error: ${err}`));
-      });
-    });
-  }
-  
-  
+  await simpleGit(rootUrl).clone(repo, cloneDir, ["--no-checkout", "--filter=tree:0"], (err) => {
```

Comment:
this looks a bit odd, why is there an error callback if you are using 'await'?

Reply:
FYI opening a PR to fix this and other functions soon!